datafusion/sql/src/expr/function.rs

```diff
@@ -149,18 +149,37 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             )?;
 
             let func_deps = schema.functional_dependencies();
-            // Find whether ties are possible in the given ordering:
-            let is_ordering_strict = order_by.iter().any(|orderby_expr| {
+            // Find whether ties are possible in the given ordering
+            let is_ordering_strict = order_by.iter().find_map(|orderby_expr| {
                 if let Expr::Sort(sort_expr) = orderby_expr {
                     if let Expr::Column(col) = sort_expr.expr.as_ref() {
-                        let idx = schema.index_of_column(col).unwrap();
-                        return func_deps.iter().any(|dep| {
+                        let idx_result = schema.index_of_column(col);
+
+                        // Propagate the error to the caller and return it immediately
+                        let idx = idx_result
```

Comment:
I think the `ok()` function call here discards the error. 

I think changing 
```rust
                        let idx = schema.index_of_column(col).unwrap();
```

To 
```rust
                        let idx = schema.index_of_column(col).ok()?;
```

accomplishes the same thing (and your test still passes)

Reply:
Thank you very much for your suggestion. Yes it's way simpler than that of mine. 

Fixed.